portfolio/src/main/webapp/script.js

```diff
@@ -17,85 +17,87 @@ let MAX_COMMENTS = 5;
+  getUserComments(userNumComments);
 }
 
 /* Retrieve user comments and display them */
 async function getUserComments(numComments=MAX_COMMENTS) {
-    try {
-        const response = await fetch(`/data?numComments=${numComments}`);
-        const data = await response.json();
+  try {
+    const response = await fetch(`/${await fetchBlobUrl()}?numComments=${numComments}`);
+    const data = await response.json();
 
-        const commentEl = document.querySelector("#user-comments");
-        if(typeof(commentEl) != 'undefined' && commentEl != null) {
-            commentEl.innerText = "";
-            // add a legend
-            commentEl.appendChild(createLegendEl());
+    const commentEl = document.querySelector("#user-comments");
+    if(typeof(commentEl) != 'undefined' && commentEl != null) {
+        commentEl.innerText = "";
 
-            for(let comment in data) {
-                commentEl.appendChild(createElement(data[comment]));
-            }
-            commentEl.style.display = "block";
+        for(let comment in data) {
+          commentEl.appendChild(createElement(data[comment]));
         }
-    } catch(err) {
-        console.log("There was an error loading comments!");
     }
+  } catch(err) {
+    console.log("There was an error loading comments!");
+  }
 }
 
 /* Delete all the comments from the server */
 async function deleteComments() {
-    try {
-        await fetch("/delete-data", {
-            method: "POST"
-        });
-    } catch(error) {
-        alert("Try again!");
-    } finally {
-        updateNumComments();
-    }
+  try {
+    await fetch("/delete-data", {
+        method: "POST"
+    });
+  } catch(error) {
+    alert("Try again!");
+  } finally {
+    updateNumComments();
+  }
 }
 
-/* Clean the fieldset children before */
-function removeAllChildren(id){
-    document.getElementById(id).innerHTML = "";
+/* Retrieve the url for the posting the comments section form and store it */
+async function fetchBlobUrl() {
+  const request = await fetch("/blobstore-upload-url");
+  // TODO get the relative link
+  return await request.text();
 }
 
-/** Creates an <p> element containing comments. */
-function createElement(comment) {
-    const pElement = document.createElement('p');
-    pElement.innerHTML = `${comment.name}: ${comment.comment}`;
-    return pElement;
+/** Set the action attribute value in the comments' form */
+function setActionAttr() {
+  const commentsForm = document.querySelector("#comment-form");
+  if(typeof(commentsForm) != 'undefined' && commentsForm != null){
+    commentsForm.actionAttrValue = fetchBlobUrl();
+  }
 }
 
-/* Creates a <legend> element */
-function createLegendEl() {
-    const legendEle = document.createElement('legend');
-    legendEle.innerHTML = "Your comments";
-    return legendEle;
+/** Creates an <p> element containing comments. */
+function createElement(comment) {
+  const pElement = document.createElement('p');
```

Comment:
Can you be consistent with naming?  I've seen 3 different variable naming styles for different elements:
- `pElement`
- `commentsForm`
- `commentEl`

Reply:
> Can you be consistent with naming? I've seen 3 different variable naming styles for different elements:
> 
> * `pElement`
> * `commentsForm`
> * `commentEl`

Thank you for highlighting this, I will correct all the names in the next repo, so that i can test everything out